clab/clab.go

```diff
@@ -100,12 +100,12 @@ func WithKeepMgmtNet() ClabOption {
 	}
 }
 
-func WithTopoFile(file, varsFile string) ClabOption {
+func WithTopoFile(file, varsFile string, deployFilter []string) ClabOption {
 	return func(c *CLab) error {
 		if file == "" {
 			return fmt.Errorf("provide a path to the clab topology file")
 		}
-		if err := c.GetTopology(file, varsFile); err != nil {
+		if err := c.GetTopology(file, varsFile, deployFilter); err != nil {
 			return fmt.Errorf("failed to read topology file: %v", err)
 		}
```

Comment:
@steiler to me it seems this has to not have a dependency on WithTopoFile if possible.
WithTopoFile just loads the topo from a file (with vars support).
The nodes filtering should be disjoint

Reply:
So it should be a WithNodeFilter, is that what you're proposing?!

Reply:
This is a great idea. It is in line with customization options we have embedded with WithXXX options